app/models/user.rb

```diff
@@ -70,10 +69,12 @@ def self.import_from_xml(xml)
           end
         end
       else
+        # Only assign divisions to new users.
+        division_name = user_data.at_css('division').text.upcase
+        attrs[:divisions] = [Division.where(name: division_name).first_or_create]
         user = User.new attrs.merge(hastus_id: hastus_id)
-        if user.valid?
-          user.save!
-            users_present << user
+        if user.save
```

Comment:
Because we have a `before_save` that sets the default password, users aren't valid until you attempt to save them.